Flaky TestHook

Seeing this for the second time:

```
=== RUN   TestHook
time="2026-04-13T17:48:47Z" level=warning msg="cannot serialize hook of type configs.FuncHook, skipping"
.....
time="2026-04-13T17:48:47Z" level=warning msg="cannot serialize hook of type configs.FuncHook, skipping"
    exec_test.go:1090: unexpected error: exit status 1
--- FAIL: TestHook (0.11s)
```

1. https://github.com/opencontainers/runc/actions/runs/24150883471/job/70477231257?pr=5222 (from https://github.com/opencontainers/runc/pull/5222).
2. https://github.com/opencontainers/runc/actions/runs/24358098981/job/71130299998 (from https://github.com/opencontainers/runc/pull/5244).

Once I saw it I opened https://github.com/opencontainers/runc/pull/5232 to be able to find out what's going on, but it is still not merged.

OK, with the help of #5232 we now know more:
```
=== RUN   TestHook
time="2026-04-18T02:22:51Z" level=warning msg="cannot serialize hook of type configs.FuncHook, skipping"
...
time="2026-04-18T02:22:51Z" level=warning msg="cannot serialize hook of type configs.FuncHook, skipping"
    exec_test.go:1098: command failed: exit status 1; stderr:
        ls: /poststart: No such file or directory
--- FAIL: TestHook (0.07s)
```

This is related to #5186. The test case runs "ls -l .... /poststart" as a container init, and now poststart hook is run *after* container init is started, so the poststart hook is racing with container init.

I don't know a good way to handle it; an OK way is to add a sleep so the poststart hook wins the race.

--- a/libcontainer/integration/exec_test.go
+++ b/libcontainer/integration/exec_test.go
@@ -784,6 +784,7 @@ func TestPassExtraFiles(t *testing.T) {
 		ExtraFiles: []*os.File{pipein1, pipein2},
 		Stdin:      nil,
 		Stdout:     &stdout,
+		Stderr:     new(strings.Builder),
 		Init:       true,
 	}
 	err = container.Run(&process)

--- a/libcontainer/integration/exec_test.go
+++ b/libcontainer/integration/exec_test.go
@@ -1075,11 +1075,14 @@ func TestHook(t *testing.T) {
 	container, err := newContainer(t, config)
 	ok(t, err)
 
-	// e.g: 'ls /prestart ...'
 	var cmd strings.Builder
 	cmd.WriteString("ls ")
 	for _, hook := range hookFiles {
-		cmd.WriteString("/" + hook + " ")
+		// The poststart hook is racing with this ls command (run as the
+		// container init), so we don't check that hook worked here yet.
+		if hook != "poststart" {
+			cmd.WriteString("/" + hook + " ")
+		}
 	}
 
 	var stdout strings.Builder
@@ -1097,6 +1100,13 @@ func TestHook(t *testing.T) {
 
 	// Wait for process
 	waitProcess(&pconfig, t)
+
+	// Check that poststart hook worked.
+	f, err = os.Open(config.Rootfs + "/poststart")
+	if err != nil {
+		t.Fatalf("poststart hook failed: %s", err)
+	}
+	f.Close()
 
 	if err := container.Destroy(); err != nil {
 		t.Fatalf("container destroy %s", err)

--- a/libcontainer/integration/execin_test.go
+++ b/libcontainer/integration/execin_test.go
@@ -421,6 +421,7 @@ func TestExecinPassExtraFiles(t *testing.T) {
 		ExtraFiles: []*os.File{pipein1, pipein2},
 		Stdin:      nil,
 		Stdout:     &stdout,
+		Stderr:     new(strings.Builder),
 	}
 	err = container.Run(inprocess)
 	ok(t, err)

